near-rust-allocator-proxy/src/allocator.rs

```diff
@@ -274,7 +274,12 @@ impl<A: GlobalAlloc> MyAllocator<A> {
             let mut addr: Option<*mut c_void> = Some(MISSING_TRACE);
             let mut chosen_i = 0;
             if layout.size() >= MIN_BLOCK_SIZE
-                || rand::thread_rng().gen_range(0, 100) < SMALL_BLOCK_TRACE_PROBABILITY
+                || NUM_ALLOCATIONS.with(|key| {
+                    let val = key.get();
+                    key.set(val + 1);
+                    val
+                }) % 100
```

Comment:
This is probably over-engineering, but it feels like it might hide some allocations. Eg, if a hot loop does exactly N allocs, where N is not coprime with 100, I thin some allocs might deterministically miss the profile. My "midndlessly copying code without thinking to much" solution here would be to copy xorshift iteration from std here: 

https://github.com/rust-lang/rust/blob/a090c8659c3be0cbc7dc93c4b2c11a9cdbf8b980/library/core/src/slice/sort.rs#L600-L607